src/binary-encoding.adoc

```diff
@@ -0,0 +1,1618 @@
+= Binary Encoding
+:toc:
+
+[[encoding_primitives]]
+== Encoding primitives
+
+[[varuint]]
+=== `VarUInt`
+
+A variable-length unsigned integer.
+
+The bytes of a ``VarUInt``s are written in
+link:https://en.wikipedia.org/wiki/Endianness:[Little Endian byte order]. This means that the first byte will contain
```

Comment:
> the first byte

do you mean "first byte_s_" or can we only represent 7-byte ints (an 8 byte VarInt would use 1 byte to describe length)?